### ISSUE TITLE
PCD Reader not seeking to correct location in Windows

### ISSUE THREAD

**Issue by dsmith8fr** [PDAL/PDAL#3424]
**Description**
PCD Reader will be off on the starting location of the data when run in Windows. This can be reproduce by using any pcd file with pdal info.
```
pdal info test_ascii.pcd -p 0
```
returns
```
(pdal info readers.pcd Error) Line 12 in './test_ascii.pcd' contains 4 fields when 5 were expected.  Ignoring.
```
with a binary file
```
pdal info test_bin.pcd -p 0
```
no error is return. However, the values are incorrect.

**Expected behavior**
Read points using PCD Reader on Windows

**System/installation information:**
```
Windows 10 x64
```

```
pdal 2.2.0 (git-version: Release)
```

<details>
  
```
$ conda list
```
# packages in environment at C:\Users\dsmith\anaconda3\envs\pdal:
#
# Name                    Version                   Build  Channel
boost-cpp                 1.74.0               h54f0996_2    conda-forge
bzip2                     1.0.8                h8ffe710_4    conda-forge
ca-certificates           2020.12.5            h5b45459_0    conda-forge
cairo                     1.16.0            hb19e0ff_1008    conda-forge
certifi                   2020.12.5        py38haa244fe_1    conda-forge
cfitsio                   3.470                h0af3d06_7    conda-forge
curl                      7.76.0               hf1763fc_0    conda-forge
expat                     2.3.0                h39d44d4_0    conda-forge
fontconfig                2.13.1            h1989441_1005    conda-forge
freetype                  2.10.4               h546665d_1    conda-forge
freexl                    1.0.6                ha8e266a_0    conda-forge
gdal                      3.2.2            py38hacca965_2    conda-forge
geos                      3.9.1                h39d44d4_2    conda-forge
geotiff                   1.6.0                h8e90983_5    conda-forge
gettext                   0.19.8.1          h1a89ca6_1005    conda-forge
hdf4                      4.2.13            h0e5069d_1004    conda-forge
hdf5                      1.10.6          nompi_h5268f04_1114    conda-forge
icu                       68.1                 h0e60522_0    conda-forge
intel-openmp              2021.2.0           h57928b3_616    conda-forge
jpeg                      9d                   h8ffe710_0    conda-forge
jsoncpp                   1.9.4                h2d74725_1    conda-forge
kealib                    1.4.14               h96bfa42_2    conda-forge
krb5                      1.17.2               hbae68bd_0    conda-forge
laszip                    3.4.3                h6538335_1    conda-forge
laz-perf                  1.4.4                ha925a31_0    conda-forge
libblas                   3.9.0                     8_mkl    conda-forge
libcblas                  3.9.0                     8_mkl    conda-forge
libcurl                   7.76.0               hf1763fc_0    conda-forge
libffi                    3.3                  h0e60522_2    conda-forge
libgdal                   3.2.2                h48ef39c_2    conda-forge
libglib                   2.68.1               h1e62bf3_0    conda-forge
libiconv                  1.16                 he774522_0    conda-forge
libkml                    1.3.0             h9859afa_1013    conda-forge
liblapack                 3.9.0                     8_mkl    conda-forge
libnetcdf                 4.7.4           nompi_h3a9aa94_107    conda-forge
libpng                    1.6.37               h1d00b33_2    conda-forge
libpq                     13.1                 h4f54205_2    conda-forge
librttopo                 1.1.0                hb340de5_6    conda-forge
libspatialite             5.0.1                h6b539a6_4    conda-forge
libssh2                   1.9.0                h680486a_6    conda-forge
libtiff                   4.2.0                hc10be44_0    conda-forge
libwebp-base              1.2.0                h8ffe710_2    conda-forge
libxml2                   2.9.10               hf5bbc77_4    conda-forge
lz4-c                     1.9.3                h8ffe710_0    conda-forge
mkl                       2020.4             hb70f87d_311    conda-forge
nitro                     2.7.dev6             h33f27b4_3    conda-forge
numpy                     1.20.2           py38h09042cb_0    conda-forge
openjpeg                  2.4.0                h48faf41_0    conda-forge
openssl                   1.1.1k               h8ffe710_0    conda-forge
pcre                      8.44                 ha925a31_0    conda-forge
pdal                      2.2.0                h56c84bb_7    conda-forge
pip                       21.0.1             pyhd8ed1ab_0    conda-forge
pixman                    0.40.0               h8ffe710_0    conda-forge
poppler                   21.03.0              h9ff6ed8_0    conda-forge
poppler-data              0.4.10                        0    conda-forge
postgresql                13.1                 h0f1a9bc_2    conda-forge
proj                      8.0.0                h1cfcee9_0    conda-forge
python                    3.8.8           h7840368_0_cpython    conda-forge
python-pdal               2.3.7            py38ha85a594_0    conda-forge
python_abi                3.8                      1_cp38    conda-forge
setuptools                49.6.0           py38haa244fe_3    conda-forge
sqlite                    3.35.4               h8ffe710_0    conda-forge
tiledb                    2.2.7                hf84e3da_0    conda-forge
tk                        8.6.10               h8ffe710_1    conda-forge
vc                        14.2                 hb210afc_4    conda-forge
vs2015_runtime            14.28.29325          h5e1d092_4    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
wincertstore              0.2             py38haa244fe_1006    conda-forge
xerces-c                  3.2.3                h0e60522_2    conda-forge
xz                        5.2.5                h62dcd97_1    conda-forge
zlib                      1.2.11            h62dcd97_1010    conda-forge
zstd                      1.4.9                h6255e5f_0    conda-forge
  
```
$ conda info
```
     active environment : pdal
    active env location : C:\Users\dsmith\anaconda3\envs\pdal
            shell level : 2
       user config file : C:\Users\dsmith\.condarc
 populated config files : C:\Users\dsmith\.condarc
          conda version : 4.9.2
    conda-build version : 3.20.5
         python version : 3.8.5.final.0
       virtual packages : __win=0=0
                          __archspec=1=x86_64
       base environment : C:\Users\dsmith\anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\dsmith\anaconda3\pkgs
                          C:\Users\dsmith\.conda\pkgs
                          C:\Users\dsmith\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\dsmith\anaconda3\envs
                          C:\Users\dsmith\.conda\envs
                          C:\Users\dsmith\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.9.2 requests/2.24.0 CPython/3.8.5 Windows/10 Windows/10.0.19041
          administrator : False
             netrc file : None
           offline mode : False

</details>

**Comment by abellgithub** [comment]
A small test file that exhibits the problem would be appreciated.

**Comment by dsmith8fr** [comment]
Both of test should demonstrate this issue.

[test_bin.zip](https://github.com/PDAL/PDAL/files/6345971/test_bin.zip)

**Comment by dsmith8fr** [comment]
I have submitted a pull request with the changes I made to resolve the problem for my system.

**Comment by abellgithub** [comment]
Specifically, this issue is that the PCD format writes all ASCII output with newlines rather than the system-appropriate line-endings.  This is noted in the current documentation.

**Comment by abellgithub** [comment]
The test files provided have NL separators rather than CR/NL as is produced by PCL on Windows. I'm guessing it shouldn't matter, but I've opened the following for clarification: https://github.com/PointCloudLibrary/pcl/issues/4724

**Comment by abellgithub** [comment]
Fixed by #3425
